compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -1698,17 +1707,19 @@ impl<'a> Context<'a> {
         len: usize,
         value: Option<AcirValue>,
     ) -> Result<(), InternalError> {
-        let databus = if self.data_bus.call_data.is_some()
-            && self.block_id(&self.data_bus.call_data.unwrap()) == array
-        {
-            BlockType::CallData
-        } else if self.data_bus.return_data.is_some()
+        let mut databus = BlockType::Memory;
+        if self.data_bus.return_data.is_some()
             && self.block_id(&self.data_bus.return_data.unwrap()) == array
         {
-            BlockType::ReturnData
+            databus = BlockType::ReturnData;
         } else {
-            BlockType::Memory
-        };
+            for array_id in self.data_bus.call_data_array() {
+                if self.block_id(&array_id) == array {
+                    databus = BlockType::CallData;
```

Comment:
Could we have simultaneously both `return_data.is_some()` and `self.block_id(&array_id) == array` (for `array_id in ..`)?

If that would be unexpected, I'd like to throw an error or at least assert so that it's visible.

Reply:
We cannot because we create new arrays.  I have added an assert.